### PR TITLE
improve java api doc

### PR DESCRIPTION
**Pull request by raulchen**
- Fix/remove incorrect contents.
- Improve phrasing.
- Change the actor example to use primitive types.

cc @salah-man

### CODE REVIEW

File: java/doc/api.rst [hunk 1]
```diff
@@ -7,51 +7,45 @@ Basic API
...
 - It must have a constructor without any parameters.
 - Any inner class must be public static.
-- It must not have a member field or method decorated using ``public static``,
-  as the semantic is undefined with multiple instances of this same class on
-  different machines
-- An actor method must be decorated using ``public`` but not ``static``
-- The other requirements are the same as for remote functions.
+- It must not have any static fields or methods, as the semantic is undefined
+  with multiple instances of this same class on different machines.
+- An actor method must be ``public``.
```
Comment (robertnishihara) [inline java/doc/api.rst]: Do all methods of the actor need to be public or just the ones that can be invoked remotely?

File: java/doc/api.rst [hunk 2]
```diff
@@ -77,15 +71,15 @@ store.
...
+locally available.
 
 ``Ray.wait``
 ~~~~~~~~~~~~
 
-Calling ``Ray.wait`` will block the current thread and wait for specified
-ray calls. It returns when at least ``numReturns`` calls are completed,
-or the ``timeout`` expires. See multi-value support for ``RayList``.
+``Ray.wait`` is used to wait for a list of ``RayObject``'s to be locally available.
+It will block the current thread until ``numReturns`` objects are ready or ``timeout``
```
Comment (robertnishihara) [inline java/doc/api.rst]: The units of the `timeout` should be clarified both here and in the function signature. Really the argument name should be `timeout_milliseconds` or something like that.

File: java/doc/api.rst [hunk 3]
```diff
@@ -114,129 +108,47 @@ as the type of an input parameter.
 ``RayList<T>``
 ^^^^^^^^^^^^^^
 
-This is a list of ``RayObject<T>``s, which inherits from ``List<T>`` in Java. It
+This is a list of ``RayObject<T>``'s, which inherits from ``List<T>`` in Java. It
```
Comment (robertnishihara) [inline java/doc/api.rst]: I think there should not be an `'` here. I wasn't completely sure, but this is my takeaway from http://grammar.ccc.commnet.edu/grammar/plurals.htm.

From the link:

*We use an apostrophe to create plural forms in two limited situations:*
- *for pluralized letters of the alphabet*
- *when we are trying to create the plural form of a word that refers to the word itself*

File: java/doc/api.rst [hunk 4]
```diff
@@ -77,15 +71,15 @@ store.
...
+These 2 methods will block the current thread until the requested data is
+locally available.
 
 ``Ray.wait``
 ~~~~~~~~~~~~
 
-Calling ``Ray.wait`` will block the current thread and wait for specified
-ray calls. It returns when at least ``numReturns`` calls are completed,
-or the ``timeout`` expires. See multi-value support for ``RayList``.
+``Ray.wait`` is used to wait for a list of ``RayObject``'s to be locally available.
```
Comment (robertnishihara) [inline java/doc/api.rst]: ``RayObject``'s -> ``RayObject``s

File: java/doc/api.rst [hunk 5]
```diff
@@ -114,129 +108,47 @@ as the type of an input parameter.
...
 
-This is a list of ``RayObject<T>``s, which inherits from ``List<T>`` in Java. It
+This is a list of ``RayObject<T>``'s, which inherits from ``List<T>`` in Java. It
 can be used as the type for both a return value and a parameter value.
 
 ``RayMap<L, T>``
 ^^^^^^^^^^^^^^^^
 
-A map of ``RayObject<T>``s with each indexed using a label with type
+A map of ``RayObject<T>``'s with each indexed using a label with type
```
Comment (robertnishihara) [inline java/doc/api.rst]: ``RayObject``'s -> ``RayObject``s

File: java/doc/api.rst (hunk 4 above)
Comment (robertnishihara) [reply java/doc/api.rst]: same change in a handful of other places in the file

File: java/doc/api.rst (hunk 1 above)
Comment (raulchen) [reply java/doc/api.rst]: Only the ones that can be invoked remotely. I'll make it more clear.

File: java/doc/api.rst (hunk 2 above)
Comment (raulchen) [reply java/doc/api.rst]: The parameter in code was already changed to `timeoutMillisecconds`, I'll update this doc accordingly.

File: java/doc/api.rst (hunk 3 above)
Comment (raulchen) [reply java/doc/api.rst]: 👍TIL.
I saw you changed one `RayObject` to "`RayObject`'s" in [this commit](https://github.com/ray-project/ray/pull/2491/commits/1abe6d11de5a6b20d29c92bcf7a35d01e255b1eb). Is that case different? Should I change it back?

Comment (robertnishihara) [reply java/doc/api.rst]: Whoops, you're right, I think it should be changed back.

File: java/doc/api.rst [hunk 6]
```diff
@@ -114,129 +108,47 @@ as the type of an input parameter.
 ``RayList<T>``
 ^^^^^^^^^^^^^^
 
-This is a list of ``RayObject<T>``s, which inherits from ``List<T>`` in Java. It
+This is a list of ``RayObject<T>``\s, which inherits from ``List<T>`` in Java. It
```
Comment (robertnishihara) [inline java/doc/api.rst]: What does the `\` do? Why not just ``RayObject<T>``s?

Comment (raulchen) [reply java/doc/api.rst]: Github won't render this correctly without the back slash.

Comment (raulchen) [reply java/doc/api.rst]: image

Comment (robertnishihara) [reply java/doc/api.rst]: Oh interesting.

**Review by robertnishihara — APPROVED** [review]

**Comment by robertnishihara** [comment]
Thanks @raulchen!

**Pull request merged**